warp/cairo-src/evm/uint256.cairo

```diff
@@ -151,12 +153,57 @@ func uint256_byte{range_check_ptr, bitwise_ptr : BitwiseBuiltin*}(a : Uint256, i
     return (res=cast((low, 0), Uint256))
 end
 
-func uint256_exp{range_check_ptr}(a : Uint256, b : Uint256) -> (res : Uint256):
-    assert 0 = 1  # not implemented yet =(
-    jmp rel 0
+func uint256_exp{range_check_ptr, bitwise_ptr : BitwiseBuiltin*}(a : Uint256, b : Uint256) -> (
+        res : Uint256):
+    alloc_locals
+    if b.low + b.high == 0:
+        return (Uint256(1, 0))
+    end
+    let (b_half) = u256_shr(b, Uint256(1, 0))
+    let (r1) = uint256_exp(a, b_half)
+    let (r2, _) = uint256_mul(r1, r1)
+    let (odd) = bitwise_and(b.low, 1)
+    if odd == 1:
+        let (r3, _) = uint256_mul(r2, a)
+        return (r3)
+    else:
+        return (r2)
+    end
+end
+
+func uint256_mulmod{range_check_ptr, bitwise_ptr : BitwiseBuiltin*}(
+        a : Uint256, b : Uint256, m : Uint256) -> (res : Uint256):
+    alloc_locals
+    let (_, ar) = uint256_unsigned_div_rem(a, m)
+    let (_, br) = uint256_unsigned_div_rem(b, m)
+    return mulmod_helper(ar, br, m)
+end
+
+func mulmod_helper{range_check_ptr, bitwise_ptr : BitwiseBuiltin*}(
+        ar : Uint256, br : Uint256, m : Uint256) -> (res : Uint256):
+    alloc_locals
+    if br.low + br.high == 0:
+        return (Uint256(1, 0))
```

Comment:
Shouldn't this return 0 instead of 1? Using only the case where `br = 0` from the beginning, it should return 0, right?

Reply:
Correct, will fix!